src/certificates_observer.py

```diff
@@ -60,56 +81,102 @@ def private_key(self) -> str:
             self._charm.app.add_secret(content={"key": private_key}, label=label)
         return private_key
 
-    @property
-    def csr(self) -> bytes:
-        """Fetch the certificate signing request.
+    def get_filebeat_csr(self, renew: bool = False) -> bytes:
+        """Fetch the certificate signing request for filebeat.
+
+        Args:
+            renew: whether to generate a new certificate signing request.
+
+        Returns: the certificate signing request.
+        """
+        return self._get_certificate_signing_request(
+            label="filebeat-csr", subject=self._charm.unit.name, renew=renew
```

Comment:
I’m just a little curious why we need to use the `tls-certificates` integration here, as a certificate with a subject like a unit name cannot be signed by a public CA. This means the only option is to use `self-signed-certificates`. But if self-signed certificates are the only option, wouldn’t it be easier to just create the certificate inside the Wazuh charm?

Reply:
OH, good point. I hadn't' noticed. I'll change the subject since we might need a certificate signed by a CA for the wazuh endpoints and I don't see the point of having 2 different relations